postgresql-plugin/src/main/java/io/cdap/plugin/postgres/PostgresDBRecord.java

```diff
@@ -49,24 +52,44 @@ public PostgresDBRecord() {
   @Override
   protected void handleField(ResultSet resultSet, StructuredRecord.Builder recordBuilder, Schema.Field field,
                              int columnIndex, int sqlType, int sqlPrecision, int sqlScale) throws SQLException {
+    ResultSetMetaData metadata = resultSet.getMetaData();
     if (isUseSchema(resultSet.getMetaData(), columnIndex)) {
       setFieldAccordingToSchema(resultSet, recordBuilder, field, columnIndex);
     } else {
+      int columnType = metadata.getColumnType(columnIndex);
+      if (columnType == Types.NUMERIC) {
+        Schema nonNullableSchema = field.getSchema().isNullable() ?
+          field.getSchema().getNonNullable() : field.getSchema();
+        int precision = metadata.getPrecision(columnIndex);
+        if (precision == 0 && Schema.Type.STRING.equals(nonNullableSchema.getType())) {
+          // When output schema is set to String for precision less numbers
+          recordBuilder.set(field.getName(), resultSet.getString(columnIndex));
+        } else if (Schema.LogicalType.DECIMAL.equals(nonNullableSchema.getLogicalType())) {
```

Comment:
Done

Reply:
was thinking something more like:

```
if (isUseSchema(resultSet.getMetaData(), columnIndex)) {
  setFieldAccordingToSchema(resultSet, recordBuilder, field, columnIndex);
  return;
}
int columnType = metadata.getColumnType(columnIndex);
if (columnType == Types.NUMERIC) {
  Schema nonNullableSchema = field.getSchema().isNullable() ?
    field.getSchema().getNonNullable() : field.getSchema();
  int precision = metadata.getPrecision(columnIndex);
  if (precision == 0 ...) {
    // When output schema is set to String for precision less numbers
    recordBuilder.set(field.getName(), resultSet.getString(columnIndex));
    return;
  }
  BigDecimal orgValue = resultSet.getBigDecimal(columnIndex);
  if (Schema.LogicalType.DECIMAL.equals(nonNullableSchema.getLogicalType() && orgValue != null) {
    BigDecimal decimalValue = ...;
    recordBuilder.setDecimal(field.getName(), decimalValue);
    return;
  }
}
setField(resultSet, ...);
```

this way there are only 2 levels of nesting instead of the current 4 levels

Reply:
done